build-support/bin/mypy.py

```diff
@@ -6,9 +6,18 @@
 
 
 def main() -> None:
-  subprocess.run(["./pants", "--backend-packages=pants.contrib.mypy", "--lint-mypy-verbose",
-                  "--lint-mypy-whitelist-tag-name=type_checked", "--tag=type_checked",
```

Comment:
We no longer rely on `--tag=type_checked`. Now we use MyPy's whitelist mechanism to handle this all.